### PR TITLE
CI job to check for BOOST_TEST macro usage

### PR DESCRIPTION
**Pull request by paulgessinger**
/cc @HadrienG2, something like this?

(had to reopen, because I'm incapable of not pushing to the upstream repo 😅 )

### CODE REVIEW

File: CI/check_boost_test_macro.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+test_string="BOOST_TEST"
+grep $test_string -n -r Tests --include "*.cpp" --include "*.hpp"
```
Comment (HadrienG2) [inline CI/check_boost_test_macro.sh:4]: I'm not sure if we actually need to search `*.hpp` here, but if we do, we probably want to search `*.ipp` too.

Comment (paulgessinger) [reply CI/check_boost_test_macro.sh:4]: In some cases I believe we have a few hpp files that included as helpers in multiple tests. That mostly got moved to separate compilation units / cpp files. 

I'll add ipp to be safe.

**Comment by paulgessinger** [comment]
Ok I think I replaced all occurrences of `BOOST_TEST`, the checker should now report green. I haven't compiled/tested the modified tests yet, I'll let the CI do that.

**Comment by paulgessinger** [comment]
Ok I redid the replacement and now it should also handle `<`, `>`, `>=` and `<=` correctly, and translate them to the corresponding macros. (This might be susceptible to some replacement errors, let's see)

File: Tests/UnitTests/Core/Geometry/CuboidVolumeBuilderTests.cpp [hunk 2]
```diff
@@ -332,21 +333,21 @@ BOOST_AUTO_TEST_CASE(CuboidVolumeBuilderTest_confinedVolumes) {
...
-        BOOST_TEST(stepResult.volume[i]->volumeName() == cvCfg1.name);
-        BOOST_TEST(stepResult.volume[i]->volumeMaterial() != nullptr);
+        BOOST_CHECK_GT(stepResult.volume[i]-, volumeName() == cvCfg1.name);
+        BOOST_CHECK_GT(stepResult.volume[i]-, volumeMaterial() != nullptr);
       } else {
         if (stepResult.position[i].x() < 2. * units::_m) {
-          BOOST_TEST(stepResult.volume[i]->volumeName() == vCfg.name);
-          BOOST_TEST(stepResult.volume[i]->volumeMaterial() == nullptr);
+          BOOST_CHECK_GT(stepResult.volume[i]-, volumeName() == vCfg.name);
+          BOOST_CHECK_GT(stepResult.volume[i]-, volumeMaterial() == nullptr);
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Geometry/CuboidVolumeBuilderTests.cpp:350]: Yeah, like this. I'll fix this.

Comment (HadrienG2) [reply Tests/UnitTests/Core/Geometry/CuboidVolumeBuilderTests.cpp]: I tend to follow a semi-automated approach for this kind of work: configure a regex-based search and replace in my text editor, but manually go through every occurence before accepting the replacement to make sure that there are no regex false positives (interactively improving my regex if need be).

**Comment by paulgessinger** [comment]
I think this works now. If there are no objections, I'll merge it.

**Pull request merged**